magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+        closest_node_before if not closest_node_before.empty else closest_node_after
+    )
+
+    if closest_node is not None and not closest_node.empty:
+        closest = closest_node.iloc[0]
+        logger.info(
+            f"\nOnly one suitable LIDAR report found for timestamp {Time(mean_subrun_timestamp, format='unix').iso} "
+            f"within the maximum allowed temporal gap. \nSkipping interpolation. Using nearest node values instead."
+            f"\n\n{closest}"
+        )
+        return (
+            closest["base_height"],
+            closest["top_height"],
+            closest["vertical_transmission"],
+        )
+
+    logger.info(
+        f"\nNo node is within the maximum allowed temporal gap for timestamp {Time(mean_subrun_timestamp, format='unix').iso}. Exiting ..."
+    )
+    return None
+
+
+def process_telescope_data(
+    dl1_params,
+    dl1_images,
+    config,
+    tel_id,
+    tel_ids,
+    camgeom,
+    focal_eff,
+    nlayers,
+    mean_subrun_zenith,
+    Hc,
+    dHc,
+    trans,
+    cmf,
+):
+    """
+    Corrects LST-1 and MAGIC data affected by a cloud presence
+
+    Parameters
+    ----------
+    dl1_params : str
+        Path to an input .h5 DL1 table with parameters
+    dl1_images : str
+        Path to an input .h5 DL1 table with images
+    config : dict
+        Configuration for the LST-1 + MAGIC analysis
+    tel_id : numpy.int16
+        LST-1 and MAGIC telescope ids
+    tel_ids : dict
+        List of LST-1 and MAGIC telescope names and ids from config file
+    camgeom : ctapipe.instrument.camera.geometry.CameraGeometry
+        An instance of the CameraGeometry class containing information about the
+        camera's configuration, including pixel type, number of pixels, rotation
+        angles, and the reference frame.
+    focal_eff : astropy.units.quantity.Quantity
+        Effective focal length
+    nlayers : astropy.units.quantity.Quantity
+        Array with heights of each cloud layer a.g.l.
+    mean_subrun_zenith : numpy.float64
+        Mean value of zenith per subran
+    Hc : astropy.units.quantity.Quantity
+        Height of the base of the cloud a.g.l.
+    dHc : astropy.units.quantity.Quantity
+        Cloud thickness
+    trans : numpy.float64
+        Transmission of the cloud
+    cmf : float
+        Multiplication factor for additional cleaning
+
+    Returns
+    -------
+    pandas.core.frame.DataFrame
+        Data frame of corrected DL1 parameters
+    """
+
+    # AC LST
+    cleaning_level = config["LST"]["tailcuts_clean"]
+    # Multiply only non-boolean numerical values
+    config_clean_LST = {
+        key: (
+            value * cmf
+            if isinstance(value, (int, float)) and not isinstance(value, bool)
+            else value
+        )
+        for key, value in cleaning_level.items()
+    }
+
+    # AC MAGIC
+    # Ignore runtime warnings appeared during the image cleaning
+    warnings.simplefilter("ignore", category=RuntimeWarning)
```

Comment:
even while this is copied from the magic_calib_do_dl1 I think it would be better to try to run without this and see if there are no issues. Because it might also hide some other warnings related to different parts of the code